crates/core/src/factories/core_fetch.rs

```diff
@@ -337,22 +315,34 @@ impl CoreFetch {
                 break;
             };
 
-            let is_peer_on_back_off = {
-                let mut lock = state.lock().unwrap();
-
-                // Do nothing if op id is no longer in the set of requests to send.
-                //
-                // Note that because this request isn't in the state, it is safe to
-                // skip the requests empty check below.
-                if !lock.requests.contains(&(op_id.clone(), peer_url.clone())) {
-                    continue;
-                }
+            // Do nothing if op id is no longer in the set of requests to send.
+            //
+            // Note that because this request isn't in the state, it is safe to
+            // skip the requests empty check below.
+            if !state
+                .lock()
+                .unwrap()
+                .requests
+                .contains(&(op_id.clone(), peer_url.clone()))
+            {
+                continue;
+            }
 
-                lock.back_off_list.is_peer_on_back_off(&peer_url)
-            };
+            // If peer URL is set as unresponsive, do not send request or re-insert
+            // request into queue, and remove all remaining requests from state.
+            let peer_url_unresponsive = peer_meta_store
+                .get_unresponsive(peer_url.clone())
+                .await
+                .is_ok_and(|maybe_value| maybe_value.is_some());
```

Comment:
Would there potentially be use in logging errors here instead of silently converting them to a boolean?

Reply:
I didn't think so. You guys do, so I'll change it.

Reply:
https://github.com/holochain/kitsune2/pull/237/commits/63a2a02dec5815830e0a7852f9da1099a3769a3d